base/src/OverlayMotionVectors.cpp

```diff
@@ -0,0 +1,113 @@
+#include <cstdint>
+#include <boost/foreach.hpp>
+extern "C"
+{
+#include <libavutil/motion_vector.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+}
+#include "OverlayMotionVectors.h"
+#include "Utils.h"
+
+class OverlayMotionVector::Detail
+{
+public:
+	Detail(OverlayMotionVectorProps props)
+	{
+	};
+	~Detail()
+	{
+	}
+
+	void setMatImg(RawImageMetadata* rawMetadata)
+	{
+		mImg = Utils::getMatHeader(rawMetadata);
+	}
+
+	void overlayMotionVectors(frame_container frames)
+	{
+		auto inRawImageFrame = Module::getFrameByType(frames, FrameMetadata::FrameType::RAW_IMAGE);
+		auto inMotionVectorFrame = Module::getFrameByType(frames, FrameMetadata::FrameType::MOTION_VECTOR_DATA);
+
+		if (inMotionVectorFrame->size() == 0 || inRawImageFrame->size() == 0)
+		{
+			return;
+		}
+
+		AVMotionVector* motionVectors = (AVMotionVector*)inMotionVectorFrame->data();
+		mImg.data = static_cast<uint8_t*>(inRawImageFrame->data());
+		for (int i = 0; i < inMotionVectorFrame->size() / sizeof(*motionVectors); i++)
+		{
+			AVMotionVector* MV = &motionVectors[i];
+
+			if (MV->motion_x > 2 || MV->motion_y > 2)
+			{
+				cv::arrowedLine(mImg, cv::Point(int(MV->src_x), int(MV->src_y)), cv::Point(int(MV->dst_x), int(MV->dst_y)), cv::Scalar(0, 255, 0), 1); // arrowedLine will help to also show the direction of motion.
+			}
+		}
+		cv::imshow("frame", mImg);
```

Comment:
Instead of showing the overlayed froma here, please send it on an output pin, if the pipeline author wants to display they will use a renderer or else they do something else @mohammedzakikochargi

Reply:
Made the overlayMotionVectors module as transform , now i'm sending the overlaid frames with outputPin